snap/quota/quota.go

```diff
@@ -832,6 +840,48 @@ func (grp *Group) NewSubGroup(name string, resourceLimits Resources) (*Group, er
 	return subGrp, nil
 }
 
+// ValidateNestingAndSnaps takes a group and verifies that it satisfies the following conditions:
+//  1. That if any parent is mixed (has both snaps and sub-groups), it must be the immediate
+//     parent group.
+//  2. If the group itself is mixed, that it has only one level of sub-grouping.
+func (grp *Group) ValidateNestingAndSnaps() error {
+	// A parent group is only allowed to contain a mixture of snaps
+	// and sub-groups if it's a direct parent. Introducing this limitation
+	// will not affect anything as we didn't allow mixing snaps and sub-groups
+	// prior to this change.
+	parent := grp.parentGroup
+	for parent != nil {
+		if len(parent.Snaps) > 0 && len(parent.SubGroups) > 0 {
```

Comment:
since our `grp` is a child of `parent`, I think that the condition `len(parent.SubGroups) > 0` will be always true.

Reply:
Really good point, thank you for catching this!